src/workerd/api/node/tests/http-client-server.js

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
```

Comment:
Maybe add a comment to note that this is running in Node.js, confused me at first not seeing `createServer` implemented.